src/classes/ERR_Handler.cls

```diff
@@ -380,9 +380,7 @@ public class ERR_Handler {
             return;
         }
 
-        if (UTIL_CustomSettingsFacade.getErrorSettings().Store_Errors_On__c) {
-            Database.insert(errorRecords, false);
-        }
+        Database.insert(errorRecords, false);
```

Comment:
This method is called from the Opp evaluation service and Allocations Settings validation on NPSP Settings and they should respect the config. See if passing a new argument and/or having a new method called from the migration batch can be used to bypass the config.